File: nginx/frontend/src/app/models/QuickSearchEntry.ts
import { AutoModerationEventTableEntry } from "./AutoModerationEventTableEntry";
import { ModCaseTable } from "./ModCaseTable";

export interface QuickSearchEntry {
    entry: ModCaseTable|AutoModerationEventTableEntry;
    createdAt: Date;
    quickSearchEntryType: number;
}

File: backend/masz/Models/QuickSearchEntry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using masz.Dtos.DiscordAPIResponses;

namespace masz.Models
{
    public interface QuickSearchEntry
    { 
        DateTime CreatedAt { get; set; }
    }
    public class QuickSearchEntry<T> : QuickSearchEntry
    {
        public T Entry { get; set; }
        public DateTime CreatedAt { get; set; }
        public QuickSearchEntryType QuickSearchEntryType { get; set; }
    }

    public enum QuickSearchEntryType
    {
        ModCase,
        AutoModeration
    }
}
